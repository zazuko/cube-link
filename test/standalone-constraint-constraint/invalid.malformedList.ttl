@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix cube: <https://cube.link/> .
@prefix observation: <https://environment.ld.admin.ch/foen/nfi/observation/max_min_undefined> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix schema: <http://schema.org/> .
@base <https://example.org/> .

<cube> a cube:Cube ;
    cube:observationConstraint <shape> ;
    cube:observationSet <observationSet> .

<observationSet> cube:observation <observation> .

<observation> a cube:Observation ;
    cube:observedBy <observer> ;
    <dimension1> "one" .


<shape> a cube:Constraint ;
    sh:targetClass cube:Observation ;
    sh:closed true ;
    sh:property
        [
            sh:path rdf:type ;
            sh:nodeKind sh:IRI ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
            sh:in [ a rdf:List ; rdf:first cube:Observation ; rdf:rest rdf:nil ] # well-formed list
        ] ;
    sh:property
        [
            sh:path cube:observedBy ; ;
            sh:nodeKind sh:IRI ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
            sh:in [ rdf:first <observer> ; rdf:rest rdf:wrong ] # malformed list
        ] ;
    sh:property
        [
            sh:datatype xsd:string ;
            sh:path <dimension1> ;
            schema:name "dimension 1" ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
            sh:in [ rdf:first "one" ] # malformed list
        ] ;
.
