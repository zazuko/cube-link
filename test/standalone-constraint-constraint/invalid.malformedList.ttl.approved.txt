@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix cube: <https://cube.link/> .

_:report a sh:ValidationReport ;
	sh:result [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Violation ;
		sh:sourceConstraintComponent sh:MinCountConstraintComponent ;
		sh:sourceShape _:b678 ;
		sh:focusNode _:b6 ;
		sh:resultPath rdf:rest ;
		sh:resultMessage "list node needs exactly one rdf:rest" ;
	], [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Violation ;
		sh:sourceConstraintComponent sh:NodeConstraintComponent ;
		sh:sourceShape [
			sh:path sh:property ;
			sh:message "sh:in needs to be a list" ;
			sh:node <https://cube.link/shape/standalone-constraint-constraint#PropertyInList> ;
		] ;
		sh:focusNode <https://example.org/shape> ;
		sh:resultPath sh:property ;
		sh:resultMessage "sh:in needs to be a list" ;
		sh:detail [
			rdf:type sh:ValidationResult ;
			sh:resultSeverity sh:Violation ;
			sh:sourceConstraintComponent sh:NodeConstraintComponent ;
			sh:sourceShape [
				sh:path sh:in ;
				sh:node <https://cube.link/shape/standalone-constraint-constraint#listnode> ;
			] ;
			sh:focusNode _:b5 ;
			sh:resultPath sh:in ;
			sh:resultMessage "Value does not have shape <https://cube.link/shape/standalone-constraint-constraint#listnode>" ;
			sh:detail [
				rdf:type sh:ValidationResult ;
				sh:resultSeverity sh:Violation ;
				sh:sourceConstraintComponent sh:MinCountConstraintComponent ;
				sh:sourceShape _:b678 ;
				sh:focusNode _:b6 ;
				sh:resultPath rdf:rest ;
				sh:resultMessage "list node needs exactly one rdf:rest" ;
			] ;
			sh:value _:b6 ;
		] ;
		sh:value _:b5 ;
	], [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Violation ;
		sh:sourceConstraintComponent sh:XoneConstraintComponent ;
		sh:sourceShape <https://cube.link/shape/standalone-constraint-constraint#restvalue> ;
		sh:focusNode rdf:wrong ;
		sh:resultMessage "rdf:rest value must be a list node or rdf:nil" ;
		sh:value rdf:wrong ;
	] ;
	sh:conforms false .

_:b678 sh:path rdf:rest ;
	sh:maxCount 1 ;
	sh:minCount 1 ;
	sh:message "list node needs exactly one rdf:rest" .
